src/component/MovieForm/MovieForm.js

```diff
@@ -1,60 +1,140 @@
- 
-  const [formData, setFormData] = useState(movieInfo);
+  const { register, handleSubmit, formState: { errors }, setValue } = useForm({
+    defaultValues: movieInfo,
+  });
 
-  const handleInputChange = (e) => {
-    setFormData({
-      ...formData,
-      [e.target.name]: e.target.value,
-    });
-  };
-
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    if (onSubmit) {
-      onSubmit(formData);
+  const handleFormSubmit = (data) => {
+    let genresList = data.genres;
+    if(!Array.isArray(genresList)) {
+      genresList = genresList.split(',');
     }
+
+    const newData = { ...data, genres: genresList, vote_average: parseInt(data.vote_average),runtime: parseFloat(data.runtime) };
+    onSubmit(newData);
   };
 
   const handleReset = () => {
-    setFormData(movieInfo);
+    if (movieInfo) {
+      Object.keys(movieInfo).forEach((key) => {
+        setValue(key, movieInfo[key]);
+      });
+    }
   };
 
+  const handleGenreInputChange = (event) => {
+    const genres = event.target.value.split(',');
+    setValue('genres', genres);
+  }
+
   return (
     <div className="movie-form_container">
-        <form onSubmit={handleSubmit} onReset={handleReset}>
+        <form onSubmit={handleSubmit(handleFormSubmit)} onReset={handleReset}>
           <div className="movie-form_columns">
-            {COLUMN_CONFIG.map(({ label, name }) => (
-              <label key={name} className="movie-form_input-label">
-                {label}
+              <label key="title" className="movie-form_input-label">
+                {TITLE}
+                <input
+                  type="text"
+                  name="title"
+                  className="movie-form_input"
+                  {...register("title", { required: 'Field is required' })}
+                />
+                  {errors.title && <span className="error-message">{errors.title.message}</span>}
+              </label>
+            <label key="release_date" className="movie-form_input-label">
+                {RELEASE_YEAR}
+                <input
+                  type="text"
+                  name="release_date"
+                  className="movie-form_input"
+                  {...register("release_date", {
+                     required: 'Field is required', 
+                     pattern: {
+                      value: /^\d{4}-\d{2}-\d{2}$/,
+                      message: 'Please enter a valid release date in YYYY-MM-DD format',
+                    }, 
+                    })}
+                />
+                  {errors.release_date && <span className="error-message">{errors.release_date.message}</span>}
+              </label>
+              <label key="poster_path" className="movie-form_input-label">
+                {MOVIE_URL}
+                <input
+                  type="text"
+                  name="poster_path"
+                  className="movie-form_input"
+                  {...register("poster_path", { 
```

Comment:
Move all the "register" methods to a separate file titled "movieFormFieldsRegistrations.js" , can be exported from there.